R/plot_tree.R

```diff
@@ -0,0 +1,331 @@
+#'   \code{lins}.
+#' @param   shapes    Shapes for the branches and leaves in the tree.
+#' @param   colours    Vector of colours.
+#' @param   colour_limits   Min and max values for the colours.
+#'
+#' @return  A \code{ggtree} object.
+
+create_noninteractive_ggtree <- function(ggtree_data,
+                                         branch_col,
+                                         lins,
+                                         lin_nodes,
+                                         lin_node_names,
+                                         shapes,
+                                         colours,
+                                         colour_limits) {
+  gtr1 <- ggtree::ggtree(
+    ggtree_data,
+    ggplot2::aes_string(colour = branch_col),
+    ladderize = TRUE,
+    right = TRUE,
+    continuous = TRUE
+  )
+
+  gtr1.1 <- gtr1 +
+    ggplot2::scale_color_gradientn(
+      name = gsub(branch_col, pattern = "_", replacement = " "),
+      colours = colours,
+      limits = colour_limits,
+      oob = scales::squish
+    ) +
+    ggplot2::geom_point(
+      ggplot2::aes_string(
+        color = branch_col,
+        size = "cluster_size",
+        shape = "as.factor(internal)"
+      ),
+      data = gtr1$data
+    ) +
+    ggplot2::scale_shape_manual(
+      name = NULL,
+      labels = NULL,
+      values = shapes
+    ) +
+    ggplot2::scale_size(
+      name = "Cluster size",
+      range = c(2, 16)
+    ) +
+    ggplot2::ggtitle(glue::glue("{Sys.Date()}, colour: {branch_col}")) +
+    ggplot2::theme(legend.position = "top")
+
+  for (i in seq_along(lins)) {
+    if (!is.na(lin_nodes[i])) {
+      gtr1.1 <- gtr1.1 +
+        ggtree::geom_cladelabel(
+          node = lin_nodes[i],
+          label = lin_node_names[i],
+          offset = .00001,
+          colour = "black"
+        )
+    }
+  }
+
+  gtr1.1
+}
+
+#' Adds data to a ggtree object to allow mouse-over tooltips etc when presented interactively
+#'
+#' @param   ggobj   A \code{ggtree} object.
+#' @param   branch_col    Scalar string. The name of a column within \code{ggobj$data} defining the
+#'   statistic under study here (`logistic_growth_rate`, `clock_outlier`).
+#' @param   sc0,cmuts   Data-frames.
+#' @param   mut_regex   Regular expression. Defines the mutations under study here.
+#'
+#' @return  A ggtree object. The \code{$data} entry has additional entries (\code{mouseover},
+#'   \code{colour_var}, \code{defmuts}, \code{allmuts}) that are used when presented interactively
+#'   by \code{ggiraph}.
+
+append_interactivity_data <- function(ggobj,
+                                      branch_col,
+                                      sc0,
+                                      cmuts,
+                                      mut_regex) {
+  # make mouseover info
+  ## standard meta data
+  ttdfs <- apply(ggobj$data, 1, FUN = function(x) {
+    z <- as.list(x)
+    lgr <- as.numeric(z$logistic_growth_rate)
+    # TODO: replace with() with explicit z$cluster_id etc
```

Comment:
Note on # TODO: A lot of the mouseover data will be removed in the next stage (this is mostly the same information that exists in the tables and it makes it difficult to click on a node when the mouseover is so large, we really only need the cluster ID), so I wouldn't worry to much about tidying up this code